drivers/flash/spi_nor.c

```diff
@@ -1365,12 +1420,18 @@ static int spi_nor_pm_control(const struct device *dev, enum pm_device_action ac
 		break;
 #else
 	case PM_DEVICE_ACTION_SUSPEND:
-		acquire_device(dev);
+		rc = acquire_device(dev);
+		if (rc < 0) {
+			return rc;
+		}
 		rc = enter_dpd(dev);
 		release_device(dev);
 		break;
 	case PM_DEVICE_ACTION_RESUME:
-		acquire_device(dev);
+		rc = acquire_device(dev);
+		if (rc < 0) {
+			return rc;
+		}
```

Comment:
Question: since acquire_device will leave the dpd, shouldn't the code below (exit_dpd) be removed.?
Also the release_device in line 1436 will bring back dpd (if CONFIG_SPI_NOR_IDLE_IN_DPD).
I think I am missing something here or some bigger rework is needed.

Reply:
`acquire_device` will only exit DPD when `SPI_NOR_IDLE_IN_DPD=y`, so `exit_dpd` can not be removed below the `acquire_device` call.

The documentation for `SPI_NOR_IDLE_IN_DPD` states that it should only be used for applications where device power management is not enabled. It doesn't make sense to have both this option and PM enabled at the same time.
If both are enabled, then yes `exit_dpd` will be called twice, but that is not a consequence of this PR and is not a functional problem, just an efficiency problem.